internal/driver/driver.go

```diff
@@ -712,3 +690,45 @@ func (d *Driver) newTemporaryOnvifClient(device models.Device) (*OnvifClient, er
 	}
 	return client, nil
 }
+
+// refreshNetworkInterfaces will attempt to retrieve the device information for the specified camera
+// and update the values in the protocol properties
+func (d *Driver) refreshNetworkInterfaces(device models.Device) error {
+	netInfo, err := d.getNetworkInterfaces(device)
+	if err != nil {
+		return err
+	}
+
+	// update device to latest version in cache to prevent race conditions
+	device, edgeXErr := sdk.RunningService().GetDeviceByName(device.Name)
+	if err != nil {
+		return edgeXErr
+	}
+
+	device.Protocols[OnvifProtocol][MACAddress] = string(netInfo.NetworkInterfaces.Info.HwAddress)
+
+	return sdk.RunningService().UpdateDevice(device)
+}
+
+// refreshDeviceInformation will attempt to retrieve the device information for the specified camera
+// and update the values in the protocol properties
+func (d *Driver) refreshDeviceInformation(device models.Device) error {
+	devInfo, err := d.getDeviceInformation(device)
+	if err != nil {
+		return err
+	}
+
+	// update device to latest version in cache to prevent race conditions
+	device, edgeXErr := sdk.RunningService().GetDeviceByName(device.Name)
+	if err != nil {
+		return edgeXErr
+	}
+
+	device.Protocols[OnvifProtocol][Manufacturer] = devInfo.Manufacturer
+	device.Protocols[OnvifProtocol][Model] = devInfo.Model
+	device.Protocols[OnvifProtocol][FirmwareVersion] = devInfo.FirmwareVersion
+	device.Protocols[OnvifProtocol][SerialNumber] = devInfo.SerialNumber
+	device.Protocols[OnvifProtocol][HardwareId] = devInfo.HardwareId
```

Comment:
same here, we should probably check if anything changed before sending a device update request.